### ISSUE TITLE
Accessibility Report of the Clippy Lints page

### ISSUE THREAD

**Issue by felix91gr** [rust-lang/rust-clippy#15604]
I just ran WAVE over https://rust-lang.github.io/rust-clippy/master/index.html to see how it fared. [Here's the report](https://wave.webaim.org/report#/https://rust-lang.github.io/rust-clippy/master/index.html)

There are a bunch of errors and alerts. They are down below, in order of importance:

# The 4 Errors

- `theme-choice` has no label
- The 3 `version-filter-selector` items have no label

The lack of labels (especially in the version filter selector) mean that understanding and using that form's item using a screen reader might require contorting over the surrounding elements to figure out what meaning and role each unlabeled form element has.

# The 1585 Contrast Errors

These are due to poor contrast, mostly between text and background color, in the elements of the page. This metric is derived from [the W3C's work](https://www.w3.org/TR/WCAG22/#contrast-minimum), and such low contrast becomes an issue for any user with impaired vision.

The large majority of these errors are the group labels:

<p align="center">  
<img width="215" height="46" alt="Two textblocks: one dark gray with white text, reading 'complexity'. The other, a warm orange with white text, reading 'warn' for warning" src="https://github.com/user-attachments/assets/b86fde27-8449-4b65-ab1e-8fd13b057de6" style="margin-left: auto; margin-right: auto;" />
</p>

White text over colored blocks needs extremely strong contrast to be legible; that's why this is considered an issue.

This however can be fixed right away with a different default Theme; one that gives more contrast by default.

# The 1621 Alerts

There are 3 alerts related to the 3 version filter selectors, which will disappear as soon as those errors are addressed. 

There are a couple of alerts related to heuristics that aren't always indicative of issues, like "Redundant Link" or having a youtube video (these are explained in detail in the report).

The rest are mostly separated into two groups.

## Small Text

Most of the Alerts are about text that is considered to be bordering on "too small" for users with any kind of visual impairment.

## Structural Issues

A couple of the remaining Alerts are related to things like having no Page Regions and having `noscript` elements, both of which impair the work of assistive technologies.

---

I hope this linting of Clippy's index of lints is useful.

I wanted to see how I could fix some of the errors, but currently my time is very limited so I'd better just post the report and hope that someone with more familiarity with this page can address them.

**Comment by felix91gr** [comment]
@GuillaumeGomez from the Zulip, I see you're working on the webpage. You might be interested in seeing this, then :)

PS: for quickly checking the contrast compliance of a pair of colors, https://webaim.org/resources/contrastchecker/ works quite well. The main source of errors right now is the low contrast at the group labels, with white text and gray or colored background. If those are fixed, the number of Contrast Errors goes from 1585 to 7

**Comment by felix91gr** [comment]
The `Very Small Text` alerts stem from the final row of text for each lint: where it says, for example,

`Applicability: MachineApplicable | Added in: 1.32.0 | ...`

The `MachineApplicable` bit and the `1.32.0` bit are too small and cause problems without zoom. This is due to them sharing the same class as the Group and Lint Level labels, which reduces their default size to 75% of what they inherit.

Since the Group and Lint Level labels are inside of a H2 heading _(BTW that is weird as well! Labeling objects should not be part of Headings)_, their size is pretty large by default, which compensates for the reduction to 75%. But for the "Applicability" and the "Added in" labels, which belong to normal paragraphs, that loss of size is too much, and they end up below the acceptable limit.

Fixing those alerts should bring their number down to the low 10s, I believe.

**Comment by GuillaumeGomez** [comment]
Accessibility is what I was planning as next step (it takes a lot of space so I needed to reduce page size first). For now I'm waiting for https://github.com/rust-lang/rust-clippy/pull/15634 and some more GUI tests to be added before doing it.

**Comment by felix91gr** [comment]
@GuillaumeGomez awesome! I'm glad to read you're thinking of accessibility ❤️ 

Ping me if there's anything I could help with :)

**Comment by GuillaumeGomez** [comment]
If you have some ideas of accessibility improvements in rustdoc, that would be welcome. :)
